spec/lib/bolt_spec/bolt_server.rb

```diff
@@ -127,6 +127,28 @@ def build_upload_request(job_id, target)
       }
     end
 
+    def build_single_file_upload_request(job_id, target)
```

Comment:
Could we just delete all but one 'file' entry in `files` for the other request? That'd save some code duplication.

Reply:
I'll take a look at removing the duplication